lib/backend.js

```diff
@@ -100,7 +100,12 @@ Backend.prototype.close = function(callback) {
   finish();
 };
 
-Backend.prototype.connect = function(connection, req) {
+Backend.prototype.connect = function(connection, req, callback) {
+  if (typeof connection === 'function') {
```

Comment:
This overload is possibly a bit contentious? Just because I've not handled all overload cases. It was mostly for convenience. We can just get rid of it if we think it's too much faff.

Reply:
Let's not do the overload, then, since we can keep it backwards-compatible without the overload.